### ISSUE TITLE
The example project in openai-java has dependency errors and cannot run properly.

### ISSUE THREAD

**Issue by zempty-zhaoxuan** [TheoKanning/openai-java#213]
The **OpenAiApiStreamExample** in the **example** project has the following error：

<img width="922" alt="Screenshot 2023-03-28 at 5 07 07 PM" src="https://user-images.githubusercontent.com/28838988/228189310-03390a76-e9e6-4a41-9884-5a607320f966.png">

<img width="713" alt="Screenshot 2023-03-28 at 5 07 42 PM" src="https://user-images.githubusercontent.com/28838988/228189515-92b7b843-a9a2-41a8-8dd9-a81196728ccf.png">

**example** project **build.gradle** should like this can solve this issue:

<img width="1159" alt="Screenshot 2023-03-28 at 5 08 29 PM" src="https://user-images.githubusercontent.com/28838988/228189837-da69a879-9c95-48e0-8cb0-08a21ceca8b2.png">

**Comment by cryptoapebot** [comment]
Confirmed, default error. 

```
> Task :example:compileJava FAILED
F:\workspace\openai-java\example\src\main\java\example\OpenAiApiStreamExample.java:35: error: cannot access Flowable
            .streamCompletionBytes(completionRequest)
                                  ^
  class file for io.reactivex.Flowable not found
1 error
```

My client build.gradle already has it.  Which did you stick it in?

**Comment by n3bul4** [comment]
The dependency for 'com.squareup.retrofit2:adapter-rxjava2:2.9.0' is already included in the build.gradle file for project service:
```
dependencies {
    api project(":client")
    api 'com.squareup.retrofit2:retrofit:2.9.0'
    implementation 'com.squareup.retrofit2:adapter-rxjava2:2.9.0'
    implementation 'com.squareup.retrofit2:converter-jackson:2.9.0'

    testImplementation(platform('org.junit:junit-bom:5.8.2'))
    testImplementation('org.junit.jupiter:junit-jupiter')
}
```
Further, project example implements project service:

```
dependencies {
    implementation project(":service")
}
```

This is actually the way gradle models dependencies between modules:(https://docs.gradle.org/current/userguide/declaring_dependencies.html). 

> Those dependencies are called project dependencies because each module is represented by a Gradle project. At runtime, the build automatically ensures that project dependencies are built in the correct order and added to the classpath for compilation.

I can't see why this error is happening provided that you haven't changed the folder structure and are using gradle with a suitable IDE.

**Comment by cryptoapebot** [comment]
Straight git checkout.  No changes.  (I also have the code in eclipse, but that's not how I am making the jars). 
Gradle on command line. 
gradle clean build test

```
PS F:\workspace\openai-java> gradle --version

------------------------------------------------------------
Gradle 8.0.2
------------------------------------------------------------

Build time:   2023-03-03 16:41:37 UTC
Revision:     7d6581558e226a580d91d399f7dfb9e3095c2b1d

Kotlin:       1.8.10
Groovy:       3.0.13
Ant:          Apache Ant(TM) version 1.10.11 compiled on July 10 2021
JVM:          19.0.2 (Oracle Corporation 19.0.2+7-44)
OS:           Windows 10 10.0 amd64
```

**Comment by zempty-zhaoxuan** [comment]
@cryptoapebot my **client** service have no this issue , I just put this dependency into **example** service's build.gradle : 
`
 implementation 'com.squareup.retrofit2:adapter-rxjava2:2.9.0'`

@n3bul4 I just clone this project into my IDEA , no any changes into code.
I guess this  issue  "Cannot access io.reactivex.Flowable" in **example** service may be due to a missing dependency that contains the io.reactivex.Flowable class during compilation.

IDEA version:
<img width="763" alt="Screenshot 2023-03-29 at 10 31 47 AM" src="https://user-images.githubusercontent.com/28838988/228414117-a7a4a0db-e896-41cd-a574-32416ea7b143.png">

gradle version:
<img width="1264" alt="Screenshot 2023-03-29 at 10 37 28 AM" src="https://user-images.githubusercontent.com/28838988/228414192-f0adaf97-50e1-4de1-b8d5-3b4a6125be32.png">

**Comment by cryptoapebot** [comment]
yes, just putting the dependency in the example/build.gradle worked, but I was curious if example depends on service, why it's not inheriting the already existing one.

**Comment by n3bul4** [comment]
@cryptoapebot  @zempty-zhaoxuan 
When I do a fresh git clone and import it into vscode I get no errors at all. Doing a gradle build from command line throws the same error. Pretty strange...as far as I have read in the gradle docs, the service project should be automatically included into the example project with all it dependencies when using this config statement:

```
implementation project(":service")
```

**Comment by n3bul4** [comment]
@xq-sj streaming support is not included in version 0.11.1.
It will be included in the next release as far as I know.

**Comment by n3bul4** [comment]
@cryptoapebot
I got a bit deeper into gradle and it seems I have found the solution to why Flowable does not get included from project service although it is a dependency of example. The problem is using 'implementation' instead of 'api' for dependency 'com.squareup.retrofit2:adapter-rxjava2'.

Declaring a dependency in the implementation configuration conceptually means it is internal to the module (it is used in the implementation but not part of the public API). Such a dependency will not be exposed to the compile classpath of consumers, though it will still be on the runtime classpath.

[Pull request](https://github.com/TheoKanning/openai-java/pull/224)
